chain/chain/src/flat_storage_resharder.rs

```diff
@@ -2009,4 +2009,116 @@ mod tests {
     fn children_catchup_after_restart() {
         children_catchup_base(true);
     }
+
+    /// The split of a parent shard shouldn't happen until the resharding block has become final.
+    #[test]
+    #[ignore]
+    fn shard_split_should_wait_final_block() {
```

Comment:
Is that test driven development in action? I love it! 
nit: Can you add a todo to remove the ignore?